### PR TITLE
Bug fixes and Param Randomization

### PR DESCRIPTION
**Pull request by Aedif**
- Re-assign Poly and Sprite textures upon `imagePath` change
- Fixed Sprite filter's alpha scaling
- Fixed `getPreset` returning presets in a dirty state
    - I noticed that applying the same preset sequentially to multiple placeables before applying it again to one that has already had that preset applied will result in some weird behaviour. I deduced that it's because of preset params being modified and then returned in a dirty state when next `getPreset` is called.
    - Solution: deepClone the params before returning them
- Added `randomized` field which randomizes params before setting the flags
   - Accepts either arrays or objects that define the range to be used to generate the value
   - Range object:
     - `val1` and `val2` define the range
     - `step` defines the increment
     - `link` allows to set another param to the same randomzied value (useful for multi-component params such as scaleX and scaleY
e.g. 
```js
  {
    filterType: "sprite",
    filterId: "randomOverlay",
    repeat: true,
    alphaDiscard: true,
    inverse: true,
    gridPadding: 1,
    translationX: 0,
    translationY: 0,
    top: true,
    randomized: {
      imagePath: [
        "modules/tokenmagic/fx/assets/distortion-1.png",
        "modules/tokenmagic/fx/assets/dots-1.png",
        "modules/tokenmagic/fx/assets/extrusion-1.png",
        "modules/tokenmagic/fx/assets/noise-2.jpg",
        "modules/tokenmagic/fx/assets/symbols-1.png",
        "modules/tokenmagic/fx/assets/waves-1.png",
      ],
      scaleX: { val1: 0.1, val2: 0.6, step: 0.01, link: 'scaleY' },
      rotation: { val1: 0, val2: 360, step: 1 },
      alpha: { val1: 0.5, val2: 1.0, step: 0.1 }
    },
  }
```

https://user-images.githubusercontent.com/7693704/219783388-cb11467a-9463-4dc1-adaa-729bdf39bcd5.mp4

### CODE REVIEW

File: tokenmagic/module/tokenmagic.js [hunk 1]
```diff
@@ -284,6 +284,36 @@ export function getPlaceableById(id, type) {
...
+    if(Array.isArray(opts) || opts.hasOwnProperty('list')){
+      const list = opts.list ?? opts;
+      params[param] = list[Math.floor(Math.random() * list.length)];
+    } else {
+      const min = Math.min(opts.val1, opts.val2);
+      const max = Math.max(opts.val1, opts.val2);
+      const stepsInRange = (max - min + (Number.isInteger(opts.step) ? 1 : 0)) / opts.step;
+      params[param] = Math.floor(Math.random() * stepsInRange) * opts.step + min;
+    }
+    if (opts.hasOwnProperty('link')) {
```
Comment (Feu-Secret) [inline tokenmagic/module/tokenmagic.js:311]: We should avoid here a division by 0 by checking `opts.step`.

**Review by Feu-Secret — APPROVED** [review]
Thanks for the PR! Approved with just a minor remark to address.

**Comment by Aedif** [comment]
Just to confirm with me now being a collaborator, would you still like to handle my commits through reviews and do the releasing yourself or do you want me to push out these sort of changes on my own?

File: tokenmagic/module/tokenmagic.js (hunk 1 above)
Comment (Aedif) [reply tokenmagic/module/tokenmagic.js]: Added a check to default `opts.step` to 1 if it's falsy. It only makes sense for step to be >=0

**Comment by Feu-Secret** [comment]
> Just to confirm with me now being a collaborator, would you still like to handle my commits through reviews and do the releasing yourself or do you want me to push out these sort of changes on my own?

Unless the changes are heavy, you can push directly to `tokenmagic-dev`. If you want to create a release, you can also. I propose to create a branch for each release, instead of using master, so that we can have pending release.

**Comment by Feu-Secret** [comment]
@Aedif Can you send me an invite on discord? We could talk directly there.

**Pull request merged**